built_in_tasks/target_tracking_task.py

```diff
@@ -555,6 +591,52 @@ def setup_screen_reset(self):
         self.bar.hide()
         self.bar.reset()
 
+    def setup_start_tracking_in(self):
+        # Revert to settable trait
+        self.limit1d = self.original_limit1d
+        # Cue successful tracking
+        self.target.cue_trial_end_success()
+
+    def setup_start_tracking_out(self):
+        # Reset target color
+        self.target.reset()
+
+    def setup_while_tracking(self):
+        # Add disturbance
+        cursor_pos = self.plant.get_endpoint_pos()
+        if self.disturbance_trial == True:
+            if self.velocity_control:
+                # TODO check manualcontrolmixin for how to implement velocity control
+                self.vel_offset = (cursor_pos + self.disturbance_path[self.frame_index])*1/self.fps
+            else: 
+                # position control
+                self.pos_offset = self.disturbance_path[self.frame_index]
+
+        # Move target and trajectory to next frame so it appears to be moving
+        self.update_frame()
+
+        # Check if the trial is over and there are no more target frames to display
+        if self.frame_index+self.lookahead >= np.shape(self.targs)[0]:
+            self.trial_timed_out = True
+
+    #### TEST FUNCTIONS ####
+    def _test_enter_target(self, time_in_state):
+        '''
+        Test if the cursor is inside the center target 
```

Comment:
maybe should rename these to enter_ramp and leave_ramp? and update the description